src/Physics/Multinucleon/EventGen/MECGenerator.cxx

```diff
@@ -1333,3 +1303,58 @@ void MECGenerator::LoadConfig(void)
     GetParamDef( "SuSA-MaxXSec-DiffTolerance", fSuSAMaxXSecDiffTolerance, 999999. );
 }
 //___________________________________________________________________________
+double MECGenerator::GetXSecMaxTlctl( const Interaction & in, 
+				      const Range1D_t Tl_range, 
+				      const Range1D_t ctl_range ) const {
+  
+  ROOT::Math::Minimizer * min = ROOT::Math::Factory::CreateMinimizer("Minuit2");
+
+  double Enu = in.InitState().ProbeE(kRfHitNucRest);
+  double LepMass = in.FSPrimLepton()->Mass();
+
+  genie::utils::mec::gsl::d2Xsec_dTCosth f(fXSecModel,&in, Enu, LepMass, -1.) ; 
+  
+  min->SetFunction( f );
+  min->SetMaxFunctionCalls(fFunctionCalls);
+  min->SetTolerance(fTolerance);
+
```

Comment:
Let's set the tolerance from file as relative, and let's evaluate the tolerance once we have the starting point. 
The set the tolerance to the minimiser.

This should make the system more flexible.